### ISSUE TITLE
confusion over how to use this module

### ISSUE THREAD

**Issue by Laubeee** [andreped/GradientAccumulator#119]
**Describe the bug**
Using `model.compile(...)` and then `model = GradientAccumulateModel(...)` as described in the [docs](https://gradientaccumulator.readthedocs.io/en/latest/background/gradient_accumulation.html) gives me an error
`RuntimeError: You must compile your model before training/testing. Use 'model.compile(optimizer, loss)'.`

using it vice versa runs without error - and it seems to be doing what it should... so perhaps just the docs are wrong?

```python
import numpy as np
import tensorflow as tf

from gradient_accumulator import GradientAccumulateModel

rows = 2048
channels = 12
features = 512

n_classes = 128

X_train = np.random.randn(rows, channels, features)
y_train = np.random.randint(0, n_classes, rows)

X_valid = np.random.randn(rows // 4, channels, features)
y_valid = np.random.randint(0, n_classes, rows // 4)

E = 10                 # number of epochs
B = 64                 # batch size
D = X_train.shape[-1]  # dimension of input sample - 768

model = tf.keras.models.Sequential([
    tf.keras.layers.Conv1D(1, 1, data_format="channels_first"),
    tf.keras.layers.Flatten(),
    tf.keras.layers.Dense(512, activation='relu'),
    tf.keras.layers.Dropout(0.2),
    tf.keras.layers.Dense(n_classes)
])
model.build(input_shape=(None, X_train.shape[1], X_train.shape[2]))

grad_accumulation_steps = 16
lr = 1e-3

optimizer=tf.keras.optimizers.Adam(learning_rate=lr)
loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
model.compile(loss=loss, optimizer=optimizer, metrics=['accuracy']) # add metrics
model = GradientAccumulateModel(accum_steps=grad_accumulation_steps, inputs=model.input, outputs=model.output)
log = model.fit(X_train[0:4096*2], y_train[0:4096*2], batch_size=B // grad_accumulation_steps, epochs=E, validation_data=(X_valid, y_valid), verbose=2)
```

**env info**
 - OS: Win11
 - Version: 0.5.2
 - Python: 3.9.16
 - TensorFlow: 2.10.1

PS: I must admit I did not downgrade my numpy version from 1.24 to 1.23 when installing this module... but that doesn't seem to be the issue.

**Comment by andreped** [comment]
Good catch, @Laubeee! That is indeed a typo.

If you check all other tests being performed under `tests/`, we do indeed compile the model after adding gradient accumulation, and not before. I will make a PR to address this.

**Comment by andreped** [comment]
Fixed in 1cfa3b70d684df224f811836dd2fc557d922dd0e.

See updated docs:
https://gradientaccumulator.readthedocs.io/en/latest/background/gradient_accumulation.html

Thank you for spotting the documentations issue, @Laubeee.

Feel free to report any other issues you may have or make PRs to address these :]
